cmd/main.go

```diff
@@ -56,75 +28,46 @@ func main() {
 
 func run() error {
 	c := config.Load()
-
 	setLogger(c.Debug)
 	c.LogFields()
 
 	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill, syscall.SIGTERM)
 	defer stop()
 
+	client, err := ethclient.Dial(c.Rpc)
+	if err != nil {
+		return fmt.Errorf("error instantiating eth client: %w", err)
+	}
+
 	group, ctx := errgroup.WithContext(ctx)
 
 	group.Go(func() error {
-		client, err := ethclient.Dial(c.Rpc)
-		if err != nil {
-			return fmt.Errorf("error instantiating eth client: %w", err)
-		}
 		s := scan.NewScanner(client, common.HexToAddress(c.ContractAddress))
 		return runScan(ctx, *c, client, s)
 	})
 
-	// Create an HTTP handler for RPC
-	handler := http.NewServeMux()
-	handler.HandleFunc("/rpc", func(w http.ResponseWriter, r *http.Request) {
-		serverCodec := jsonrpc.NewServerCodec(&httpReadWriteCloser{r.Body, w})
-		w.Header().Set("Content-type", "application/json")
-		rpcErr := rpc.ServeRequest(serverCodec)
-		if rpcErr != nil {
-			slog.Warn("error while serving JSON request", "err", rpcErr)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-	})
-
-	// Create an HTTP server with timeouts
-	server := &http.Server{
-		Addr:         fmt.Sprintf(":%d", c.Port),
-		Handler:      handler,
-		ReadTimeout:  5 * time.Second,
-		WriteTimeout: 10 * time.Second,
-		IdleTimeout:  120 * time.Second,
-	}
-
-	slog.Info("starting universal node RPC server", "port", c.Port)
-
-	group.Go(func() error {
-		<-ctx.Done()
-		slog.Info("shutting down the RPC server...")
-		shutdownCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-		defer cancel()
-		if shutdownErr := server.Shutdown(shutdownCtx); shutdownErr != nil {
-			return fmt.Errorf("error shutting down the RPC server: %w", shutdownErr)
-		}
-		return nil
-	})
 	group.Go(func() error {
-		sys := new(System)
-		err := rpc.Register(sys)
+		rpcServer, err := server.NewServer(
+			server.WithEthService(client),
+			server.WithNetService(c.ChainID),
+			server.WithSystemHealthService(),
+		)
 		if err != nil {
-			return fmt.Errorf("error registering RPC service: %w", err)
-		}
-		if srvErr := server.ListenAndServe(); srvErr != nil && srvErr != http.ErrServerClosed {
-			return srvErr
+			slog.Error("failed to create RPC server: %v", err)
+			return err
 		}
-		return nil
+		addr := fmt.Sprintf("0.0.0.0:%v", c.Port)
+		slog.Info("Starting RPC server", "listenAddress", addr)
```

Comment:
Please log messages with the first lowercase letter for sake of consistency, unless they are acronyms, like `RPC`. Thanks